general-superstaq/general_superstaq/pretty_printing_test.py

```diff
@@ -0,0 +1,292 @@
+    compiler_output_repr = (
+        "CompilerOutput([cirq.Circuit([\n"
+        "    cirq.Moment(\n"
+        "        cirq.X(cirq.LineQubit(0)),\n"
+        "    ),\n"
+        "]), cirq.Circuit([\n"
+        "    cirq.Moment(\n"
+        "        cirq.X(cirq.LineQubit(0)),\n"
+        "    ),\n"
+        "]), cirq.Circuit([\n"
+        "    cirq.Moment(\n"
+        "        cirq.X(cirq.LineQubit(0)),\n"
+        "    ),\n"
+        "]), cirq.Circuit([\n"
+        "    cirq.Moment(\n"
+        "        cirq.X(cirq.LineQubit(0)),\n"
+        "    ),\n"
+        "]), cirq.Circuit([\n"
+        "    cirq.Moment(\n"
+        "        cirq.X(cirq.LineQubit(0)),\n"
+        "    ),\n"
+        "])], [{cirq.LineQubit(0): cirq.LineQubit(0)}, {cirq.LineQubit(0): cirq.LineQubit(0)}, "
+        "{cirq.LineQubit(0): cirq.LineQubit(0)}, {cirq.LineQubit(0): cirq.LineQubit(0)}, "
+        "{cirq.LineQubit(0): cirq.LineQubit(0)}], [Schedule((0, Play(Drag(duration=160, sigma=40, "
+        "beta=-1.040499642039664, amp=0.4033317218845395, angle=0.0, name='Xp_d0'), "
+        "DriveChannel(0), name='Xp_d0')), name=\"sched1\"), Schedule((0, Play(Drag(duration=160, "
+        "sigma=40, beta=-1.040499642039664, amp=0.4033317218845395, angle=0.0, name='Xp_d0'), "
+        "DriveChannel(0), name='Xp_d0')), name=\"sched2\"), Schedule((0, Play(Drag(duration=160, "
+        "sigma=40, beta=-1.040499642039664, amp=0.4033317218845395, angle=0.0, name='Xp_d0'), "
+        "DriveChannel(0), name='Xp_d0')), name=\"sched3\"), Schedule((0, Play(Drag(duration=160, "
+        "sigma=40, beta=-1.040499642039664, amp=0.4033317218845395, angle=0.0, name='Xp_d0'), "
+        "DriveChannel(0), name='Xp_d0')), name=\"sched4\"), Schedule((0, Play(Drag(duration=160, "
+        "sigma=40, beta=-1.040499642039664, amp=0.4033317218845395, angle=0.0, name='Xp_d0'), "
+        "DriveChannel(0), name='Xp_d0')), name=\"sched5\")], None, None, None)"
+    )
+    circuit_reprs = [
+        "cirq.Circuit([\n    cirq.Moment(\n        cirq.X(cirq.LineQubit(0)),\n    ),\n])"
+    ] * 5
+    circuit_drawings = ["0: ───X───"] * 5
+    assert gss.pretty_printing.pretty_print_compiler_output(
+        compiler_output_repr, circuit_reprs, circuit_drawings
+    ) == (
+        "CompilerOutput(\n"
+        "    [\n"
+        "        0: ───X───,\n"
+        "        \n"
+        "        0: ───X───,\n"
+        "        \n"
+        "        0: ───X───,\n"
+        "        \n"
+        "        0: ───X───,\n"
+        "        \n"
+        "        0: ───X───,\n"
+        "    ],\n"
+        "    [\n"
+        "        {cirq.LineQubit(0): cirq.LineQubit(0)},\n"
+        "        {cirq.LineQubit(0): cirq.LineQubit(0)},\n"
+        "        {cirq.LineQubit(0): cirq.LineQubit(0)},\n"
+        "        {cirq.LineQubit(0): cirq.LineQubit(0)},\n"
+        "        {cirq.LineQubit(0): cirq.LineQubit(0)},\n"
+        "    ],\n"
+        "    [\n"
+        "        Schedule(\n"
```

Comment:
it would be nice to abbreviate things that are this long - i think one of the goals of adding a pretty repr is to make it so e.g. typing `compiler_output` in a console/notebook just prints something manageable and (somewhat) succinct

(i think `textwrap.shorten` does something like this?)